### ISSUE TITLE
Watir will hang forever if you attempt to click a hidden element

### ISSUE THREAD

**Issue by pjdjr** [watir/watir#653]
## Meta -
Watir Version: 6.8.1
Selenium Version: 3.5.1
Browser Version: Chrome 61.0.3163.79
Browser Driver Version: Chromedriver 2.32
OS Version: Linux Mint 17.3

## Expected Behavior -
Attempting to click a hidden element should probably not be possible. I can't remember the previous behavior or if I ever tried it (I stumbled across this by unfortunate accident). The timeout should be no more than the default timeout (30 seconds) in any case.

## Actual Behavior -
Watir will hang (seemingly) forever.

**In my test log I see**

[WARN] 03:11:10 PM :: Watir::Exception::UnknownObjectException exception occurred in session: unable to locate element: #<Watir::Button: located: false; {:xpath=>"//div[@aria-describedby='deleteTriggerDialog']/div[3]/div/button", :tag_name=>"button"}>
[WARN] 03:11:10 PM :: exception stacktrace:
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/elements/element.rb:585:in `assert_element_found'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/elements/element.rb:580:in `assert_exists'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/elements/element.rb:677:in `element_call'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/elements/element.rb:413:in `enabled?'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait.rb:46:in `block in until'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait.rb:89:in `block in run_with_timer'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait/timer.rb:22:in `block in wait'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait/timer.rb:21:in `loop'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait/timer.rb:21:in `wait'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait.rb:88:in `run_with_timer'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait.rb:45:in `until'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/wait.rb:125:in `wait_until'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/elements/element.rb:547:in `wait_for_enabled'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/elements/element.rb:677:in `element_call'
/home/a/.rvm/gems/ruby-2.3.0/gems/watir-6.8.1/lib/watir/elements/element.rb:124:in `click'

**In watir.log I see two lines repeated over and over and over:**

2017-09-19 15:11:08 INFO Watir -> `#<Watir::Button: located: true; {:xpath=>"//div[@aria-describedby='deleteTriggerDialog']/div[3]/div/button", :tag_name=>"button"}>#enabled?` after `#assert_exists` (as a prerequisite for a previously specified execution)
2017-09-19 15:11:09 INFO Watir <- `#<Watir::Button: located: true; {:xpath=>"//div[@aria-describedby='deleteTriggerDialog']/div[3]/div/button", :tag_name=>"button"}>#enabled?` has been completed
2017-09-19 15:11:09 INFO Watir -> `#<Watir::Button: located: true; {:xpath=>"//div[@aria-describedby='deleteTriggerDialog']/div[3]/div/button", :tag_name=>"button"}>#enabled?` after `#assert_exists` (as a prerequisite for a previously specified execution)
2017-09-19 15:11:09 INFO Watir <- `#<Watir::Button: located: true; {:xpath=>"//div[@aria-describedby='deleteTriggerDialog']/div[3]/div/button", :tag_name=>"button"}>#enabled?` has been completed
2017-09-19 15:11:09 INFO Watir -> `#<Watir::Button: located: true; {:xpath=>"//div[@aria-describedby='deleteTriggerDialog']/div[3]/div/button", :tag_name=>"button"}>#enabled?` after `#assert_exists` (as a prerequisite for a previously specified execution)
2017-09-19 15:11:09 INFO Watir <- `#<Watir::Button: located: true; {:xpath=>"//div[@aria-describedby='deleteTriggerDialog']/div[3]/div/button", :tag_name=>"button"}>#enabled?` has been completed
.
.
.

**And in chromedriver.log I see this series of lines repeated over and over and over:**

[1752.862][INFO]: Waiting for pending navigations...
[1752.863][INFO]: Done waiting for pending navigations. Status: ok
[1752.867][INFO]: Waiting for pending navigations...
[1752.868][INFO]: Done waiting for pending navigations. Status: ok
[1752.868][INFO]: RESPONSE IsElementEnabled true
[1752.869][INFO]: COMMAND IsElementEnabled {
   "id": "0.5948328318003362-21"
}
[1752.869][INFO]: Waiting for pending navigations...
[1752.869][INFO]: Done waiting for pending navigations. Status: ok
[1752.873][INFO]: Waiting for pending navigations...
[1752.873][INFO]: Done waiting for pending navigations. Status: ok
[1752.873][INFO]: RESPONSE IsElementEnabled true
[1752.874][INFO]: COMMAND ClickElement {
   "id": "0.5948328318003362-21"
}
[1752.874][INFO]: Waiting for pending navigations...
[1752.875][INFO]: Done waiting for pending navigations. Status: ok
[1752.883][INFO]: Waiting for pending navigations...
[1752.884][INFO]: Done waiting for pending navigations. Status: ok
[1752.884][INFO]: RESPONSE ClickElement element not visible
  (Session info: chrome=61.0.3163.91)
[1752.885][INFO]: COMMAND IsElementEnabled {
   "id": "0.5948328318003362-21"
}

## Steps to reproduce -
**A simple way to reproduce is to create this web page:**

```
<html>
<body>
<button hidden>Yes</button>
<button>No</button>
</body>
</html>
```

**Use Watir to navigate to the above page and**
`browser.button(:text, "Yes").click`

For me, Watir tries and tries to click. After 4+ hours Watir did not time out.

**Comment by kapoorlakshya** [comment]
Try Watir 6.8.4. I believe this was fixed recently.

**Comment by titusfortner** [comment]
I fixed disabled and read only, hidden might still be broken. :-)

**Comment by pjdjr** [comment]
Good news! I updated my Watir gem this morning and this does seem to be fixed in 6.8.4. Searching for a hidden element now times out after 30 seconds.
Thanks!
